var/spack/repos/builtin/packages/charm/package.py

```diff
@@ -47,63 +50,109 @@ class Charm(Package):
 
     # Support OpenMPI; see
     # <https://charm.cs.illinois.edu/redmine/issues/1206>
-    patch("mpi.patch")
+    # Patch is no longer needed in versions 6.8.0+
+    patch("mpi.patch", when="@:6.7.1")
+
     # Ignore compiler warnings while configuring
     patch("strictpass.patch")
 
+    # Build targets
+    # "target" is reserved, so we have to use something else.
+    variant(
+        "build-target",
+        default="LIBS",
+        values=("charm++", "AMPI", "LIBS"),
+        description="Specify the target to build"
+    )
+
     # Communication mechanisms (choose exactly one)
-    # TODO: Support Blue Gene/Q PAMI, Cray GNI, Cray shmem, CUDA
     variant(
-        'backend',
-        default='mpi',
-        values=('mpi', 'multicore', 'net', 'netlrts', 'verbs'),
-        description='Set the backend to use'
+        "backend",
+        default="mpi",
+        values=("mpi", "multicore", "netlrts", "verbs", "gni",
+                "ofi", "pami", "pamilrts"),
+        description="Set the backend to use"
     )
 
     # Other options
-    # Something is off with PAPI -- there are build errors. Maybe
-    # Charm++ expects a particular version?
     variant("papi", default=False, description="Enable PAPI integration")
+    variant("syncft", default=False, description="Compile with Charm++ fault tolerance support")
     variant("smp", default=True,
             description=(
                 "Enable SMP parallelism (does not work with +multicore)"))
     variant("tcp", default=False,
             description="Use TCP as transport mechanism (requires +net)")
+    variant("omp", default=False, description="Support for the integrated LLVM OpenMP runtime")
+    variant("pthreads", default=False, description="Compile with pthreads Converse threads")
+    variant("cuda", default=False, description="Enable CUDA toolkit")
+
     variant("shared", default=True, description="Enable shared link support")
+    variant("production", default=True, description="Build charm++ with all optimizations")
+    variant("tracing", default=False, description="Enable tracing modules")
 
-    # Note: We could add variants for AMPI, LIBS, bigemulator, msa, Tau
+    variant("destination", description="Build charm++ inside DIR, by default the destination is <version>")
```

Comment:
Shouldn't the default be `prefix`? Does this allow you to install the package in a location other that Spack's installation prefix? This will likely break other packages that depend on this one.

Reply:
After discussion, it seems ok to just remove this option if it could cause problems with other packages, it is not a strict requirement. I'll make another commit with the changes.